ctapipe/calib/camera/charge_extraction.py

```diff
@@ -0,0 +1,281 @@
+from abc import abstractmethod
+import numpy as np
+from traitlets import Int, Unicode
+
+from ctapipe.core import Component
+
+
+def all_subclasses(cls):
+    return cls.__subclasses__() + [g for s in cls.__subclasses__()
+                                   for g in all_subclasses(s)]
+
+
+class ChargeExtractor(Component):
+    name = 'ChargeExtractor'
+
+    def __init__(self, waveforms, parent, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+
+        self.waveforms = waveforms
+        self.nchan, self.npix, self.nsamples = waveforms.shape
+
+    @abstractmethod
+    def extract_charge(self):
+        """ Calls the relevant functions to fully extract the charge for the
+        particular extractor """
+
+
+class Integrator(ChargeExtractor):
+    def __init__(self, waveforms, parent, **kwargs):
+        super().__init__(waveforms, parent=parent, **kwargs)
+
+        self.__window_width = np.zeros((self.nchan, self.npix), dtype=np.intp)
+        self.__window_start = np.zeros((self.nchan, self.npix), dtype=np.intp)
+
+        self.integration_window = np.zeros_like(self.waveforms, dtype=bool)
+        self.windowed_waveforms = None
+        self.integrated_waveform = None
+        self.peakpos = None
+
+    @property
+    def window_width(self):
```

Comment:
Using double underscores will make it impossible to inherit these values. Is it intentional? 

See https://docs.python.org/2/tutorial/classes.html#private-variables-and-class-local-references